### ISSUE TITLE
istio rewrite http probe adds duplicate header 

### ISSUE THREAD

**Issue by winterTTr** [istio/istio#31169]
**Bug description**
Istio will by default [enable the http probe rewrite when enable the mTLS.](https://istio.io/latest/docs/ops/configuration/mesh/app-health-check/#liveness-and-readiness-probes-using-the-http-request-approach). However, when the pod probe definition includes the extra `httpHeaders` section, such as 
```
        readinessProbe:
          failureThreshold: 3
          httpGet:
            httpHeaders:
            - name: bbbb
              value: "00000000"
            path: /http
            port: 5000
            scheme: HTTP
          periodSeconds: 15
          successThreshold: 1
          timeoutSeconds: 5
```
The underlying service container will receive multi header of "bbbb", not just single one.  Such as, this is one of the log about the detail request header which service container received:
```
User-Agent: kube-probe/1.18
Accept-Encoding: gzip
Bbbb: 00000000
Bbbb: 00000000
Connection: close
```

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
There should be one single header received, not duplicate header. 

**Steps to reproduce the bug**
Deploy any deployment on an mTLS enabled istio environment, add an extra http readiness probe with `httpHeaders`. 

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
```
> istioctl version --remote
client version: 1.8.2
control plane version: 1.8.2
data plane version: 1.8.2 (4 proxies)
```
```
> kubectl version --short
Client Version: v1.18.6
Server Version: v1.18.6
```

**How was Istio installed?**
Using IstioOperator

**Environment where the bug was observed (cloud vendor, OS, etc)**
Azure AKS, Linux ubuntu

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

**Comment by bianpengyuan** [comment]
Dupe of https://github.com/istio/istio/issues/28466. Fix will be shipped with 1.8.4.

**Comment by bianpengyuan** [comment]
Will close since this is already fixed. Please reopen if 1.8.4 does not fixed the issue. Thanks!
